packages/core/src/execution/dispatch/utils.ts

```diff
@@ -31,7 +32,7 @@ function resolveFromContext(
 
   const entry = context.get(arg.vertexId);
 
-  if (!entry) {
+  if (entry === undefined || entry === null) {
```

Comment:
Not-blocking: I think in general returning both `undefined` and `null` can signal that you should be returning a special type from that function.

Reply:
Taking a look at the root cause of the entry possibly being undefined/null, the undefined comes from the `map.get`, so undefined is always a possibility. The value of the map can sometimes be null, as the internal execution state result can be `null`, the value it will have before it is populated by a dispatch.

In general I prefer using `null` to `undefined` for values that have not yet been set, but will be set, to avoid the confusion of doing a lookup with the wrong property and getting undefined and doing a lookup early in the process and the value not being set yet. But that might just be my javascript roots showing through.

The execution state that has result possibly null, is entirely internal and controlled by us, so I have no problem switching to undefined in this case to clean up the check (I have pushed a commit to that effect.

Reply:
Thanks for the explanation, and sorry, as I was confused here. I didn't realize `context` was a builtin `Map`.